starport/services/network/reward.go

```diff
@@ -0,0 +1,47 @@
+package network
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	rewardtypes "github.com/tendermint/spn/x/reward/types"
+	"github.com/tendermint/starport/starport/pkg/events"
+	"github.com/tendermint/starport/starport/services/network/networktypes"
+)
+
+// SetReward set a chain reward
+func (n Network) SetReward(launchID, lastRewardHeight uint64, coins sdk.Coins) error {
+	n.ev.Send(events.New(
+		events.StatusOngoing,
+		fmt.Sprintf("Setting reward %s to the chain %d at height %d",
+			coins.String(),
+			launchID,
+			lastRewardHeight,
+		),
+	))
+
+	msg := rewardtypes.NewMsgSetRewards(
+		n.account.Address(networktypes.SPN),
+		launchID,
+		lastRewardHeight,
+		coins)
+	res, err := n.cosmos.BroadcastTx(n.account.Name, msg)
+	if err != nil {
+		return err
+	}
+
+	var setRewardRes rewardtypes.MsgSetRewardsResponse
+	if err := res.Decode(&setRewardRes); err != nil {
+		return err
+	}
+
+	n.ev.Send(events.New(events.StatusDone,
+		fmt.Sprintf(
+			"%s will be distributed to validators at height %d. The chain %d is now an incentivized testnet",
```

Comment:
It is also possible to add rewards during publish (https://github.com/tendermint/starport/issues/2076).

Here user just may want to update the rewards. In this case it may not be sensible to tell that "The chain %d is now an incentivized testnet"?

Maybe we can check if the reward set before and if not we can print this message?

Reply:
> It is also possible to add rewards during publish (#2076).
> 
> Here user just may want to update the rewards. In this case it may not be sensible to tell that "The chain %d is now an incentivized testnet"?

yes, sure, I will implement it in another PR after we merge this one

> Maybe we can check if the reward set before and if not we can print this message?

The reward will always be set. The SPN cannot accept empty reward coins into `MsgSetReward`. This is something we can discuss. Maybe we can use the same method to delete a reward from a chain if we allow empty rewards and print different log messages in the starport CLI. What do you think, @ilgooz @lubtd?

Reply:
> Maybe we can check if the reward set before and if not we can print this message?

I mean we can check if the reward set before by another call to reward-set or during publish.

Reply:
I will add a better log after we merge this PR: https://github.com/tendermint/spn/pull/542
with this one, we can have more info in the message response.